link/tracepoint_test.go

```diff
@@ -38,6 +39,10 @@ func TestTracepoint(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	pe, ok := tp.(*perfEvent)
```

Comment:
we do validate the program type on `attach`:

```
	case tracepointEvent:
            ...
```
so in theory it's important that `Tracepoint` has the correct event type; in this specific case it's the zero value of the enum so it will pass anyway even if not set but IMO we should treat it as every other perfEvent abstraction

Reply:
> we do validate the program type on `attach`

Exactly, so we should test the _behaviour_ of that validation by passing in a prog with the wrong type, that's what matters the most!

I wouldn't reach across the `Link` interface boundary here. After all, we just wrote the `typ` value to the underlying struct, we're pretty sure it will come out okay on the other side. :smile:

Reply:
I see what you mean, will change it